core/cthreads.c

```diff
@@ -0,0 +1,291 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "cthreads.h"
+
+#if __unix__
```

Comment:
Here's where you'd want to use that silly `OS_UNIX` macro after including `osname.h`